### PR TITLE
fix(DHIS2-13915): show spinner when an app is being installed

### PR DESCRIPTION
**Pull request by kabaros**
fixes [DHIS2-13915](https://dhis2.atlassian.net/browse/DHIS2-13915) and  [DHIS2-15586](https://dhis2.atlassian.net/browse/DHIS2-15586).

**DHIS2-13915**

This adds a spinner to the install button in the main app details and the version list
- the UX for the version list spinner is not great with the button resizing and pushing the table, but it's better than the previous complete lack of feedback - I am open to easy workarounds that improves this UX though!

appmanagement-spinner

appmangement-versions-loader

**DHIS2-13915**

This PR also fixes [DHIS2-15586](https://dhis2.atlassian.net/browse/DHIS2-15586). This is a workaround in the absence of an ID we can rely on, it matches the app with the installed apps using the name + developer email. The chances of these conflicting are low, and I think this is better than the current situation where users can't tell at all if an app was installed or not (check video in Jira ticket).

appmanagement-fix-after

[DHIS2-13915]: https://dhis2.atlassian.net/browse/DHIS2-13915?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DHIS2-15586]: https://dhis2.atlassian.net/browse/DHIS2-15586?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DHIS2-15586]: https://dhis2.atlassian.net/browse/DHIS2-15586?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/AppDetails/ManageInstalledVersion.js [hunk 1]
```diff
@@ -74,10 +80,22 @@ export const ManageInstalledVersion = ({
...
+                            <>
+                                {i18n.t('Installing...')}
+                                <CircularLoader small />
+                            </>
+                        )}
+                        {!isInstalling
+                            ? canUpdate
+                                ? i18n.t('Update to latest version')
+                                : i18n.t('Install')
+                            : ''}
```
Comment (Mohammer5) [inline src/components/AppDetails/ManageInstalledVersion.js:98]: The button has an `icon` property, then it'd be in line with all other icon-buttons.
Also applies to buttons in files below

File: src/pages/AppHubApp/AppHubApp.js [hunk 2]
```diff
@@ -47,6 +47,10 @@ export const AppHubApp = ({ match }) => {
             </NoticeBox>
         )
     }
+
+    // ToDo: This check here is the cause of the bug https://dhis2.atlassian.net/browse/DHIS2-15586
+    // custom apps seems to not have an app_hub_id, when these are surfaced then this should be resolved
```
Comment (Mohammer5) [inline src/pages/AppHubApp/AppHubApp.js:52]:
```suggestion
    // custom apps seem to not have an app_hub_id, when these are surfaced then this should be resolved
```

**Review by Mohammer5 — APPROVED** [review]
Approving this so I'm not blocking the PR in case my feedback won't be addressed until next week when I'm on vacation

File: src/components/AppDetails/Versions.js [hunk 3]
```diff
@@ -120,28 +125,34 @@ const VersionsTable = ({
...
-                                        <CircularLoader small />
-                                    </>
-                                )}
-                                {!isVersionInstalling
-                                    ? version.version === installedVersion
-                                        ? i18n.t('Installed')
-                                        : i18n.t('Install')
-                                    : ''}
-                            </Button>
+                            <div className={styles.tableActions}>
```
Comment (kabaros) [inline src/components/AppDetails/Versions.js:128]: this was necessary otherwise the two buttons were not aligned properly when the loader is shown

**Comment by kabaros** [comment]
@tomzemp I encountered the error with dhis/prop-types - this was caused by this project requiring an older version that exposed all PropTypes from `prop-types` library, but we stopped doing that since v3 - I could have resolved it with adding a resolutions to package.json, but chose to just drop the use of `@dhis2/prop-types` since we only use it now for special types.

File: src/components/AppDetails/Versions.js (hunk 3 above)
Comment (Mohammer5) [reply src/components/AppDetails/Versions.js:128]: I think [this component](https://ui.dhis2.nu/demo/?path=/story/button-strip--default) is supposed to do the same thing. But I don't know if it works with the link

Comment (kabaros) [reply src/components/AppDetails/Versions.js]: yeah that worked - updated it. Thanks @Mohammer5

**Comment by tomzemp** [comment]
makes sense @kabaros. no issues running for me now 🙏

**Review by Mohammer5 — APPROVED** [review]

**Review by tomzemp — APPROVED** [review]
looks good 🚀

**Pull request merged**